### PR TITLE
feat(sitemap): allow ESI in XML, pass (owner,repo,ref) to cgi-bin

### LINKED ISSUE #312: CGI actions should be provided with owner, repo and ref
These should be provided as parameters `__hlx_owner`, `__hlx_repo` and `__hlx_ref`.

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #312

### CODE REVIEW

File: layouts/fastly/helix.vcl [hunk 1]
```diff
@@ -726,8 +733,8 @@ sub hlx_fetch_static {
       # Keep the redirect around for a short bit, to prevent thundering herd
       set beresp.cacheable = true;
       set beresp.ttl = 5s;
+      return(deliver);
     }
-    return(deliver);
```
Comment (dominique-pfister) [inline layouts/fastly/helix.vcl]: returning `deliver` from here will stop processing `vcl_fetch` and the ESI handling never gets executed.

File: layouts/fastly/helix.vcl [hunk 2]
```diff
@@ -986,7 +993,7 @@ sub vcl_recv {
   if (req.http.X-From-Edge) {
     # Request came from a Fastly POP, send the raw response without ESI processing
     set req.esi = false;
-  } elseif ( req.url.ext == "html" ) {
+  } elseif ( req.url.ext == "html" || req.url.ext == "xml") {
```
Comment (dominique-pfister) [inline layouts/fastly/helix.vcl]: Handle ESI in XML as well

File: layouts/fastly/helix.vcl [hunk 3]
```diff
@@ -557,19 +557,26 @@ sub hlx_type_cgi {
...
 
   set req.http.X-Backend-URL = "/api/v1/web"
-    + "/" + var.namespace // i.e. /trieloff
+    + "/" + var.namespace
     + "/" + var.package
     // looks like cgi-bin-hello-world for /cgi-bin/hello-world.js
     + "/cgi-bin-" + var.script
-    // we keep the complete query string
-    + "?" + req.url.qs;
+    + "?__hlx_owner=" + req.http.X-Github-Static-Owner
```
Comment (tripodsan) [inline layouts/fastly/helix.vcl]: why static?

Comment (dominique-pfister) [reply layouts/fastly/helix.vcl]: If it should be `code` instead, I don't see it available in the Fastly tables...

**Pull request merged**